lib/fauxpaas/artifact_builder.rb

```diff
@@ -38,5 +39,36 @@ def add_reference(ref, path)
         .write
     end
 
+		# Set the mode bits on all files in the Artifact
+		# @param path [Pathname]
+		# @param asset_dir [String]
+		def set_access_control(path, asset_dir = "public")
+			asset_path = path/asset_dir 
+
+      # Set permissions on artifact
+      if Dir.exist?(path)
+        Find.find(path) do |found_path|
+          if File.directory?(found_path)
+            FileUtils.chmod(2770, found_path)
+          else
+            FileUtils.chmod(0660, found_path)
+          end
+        end
+      end
+
+      # If asset directory exists set permissions on it also
+      if Dir.exists?(asset_path)
```

Comment:
I'd extract this as a separate function, which also allows us to put the `Dir.exist?` check in a "guard clause", which reduces one level of nesting:

```ruby

def chmod_path(base_dir, file_mode, dir_mode)
  return unless Dir.exist?(base_dir)

  Find.find(base_dir) do |path|
  if File.directory?(path)
    FileUtils.chmod(dir_mode, path)
  else
    FileUtils.chmod(file_mode, path)
  end
 
end
```

Reply:
Then you could rewrite `set_access_control` as:

```ruby

def set_access_control(path)
  chmod_path(path,file_mode: 0660, dir_mode: 2770)
  chmod_path(path/"public",file_mode: 0664, dir_mode: 2775)
end
```

Reply:
Yup. Looking into that. And thanks.